projects/batfish-common-protocol/src/test/java/org/batfish/datamodel/pojo/AnalysisTest.java

```diff
@@ -0,0 +1,90 @@
+package org.batfish.datamodel.pojo;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.testing.EqualsTester;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.batfish.common.BatfishException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link Analysis}. */
+@RunWith(JUnit4.class)
+public class AnalysisTest {
+
+  @Rule public ExpectedException _thrown = ExpectedException.none();
+
+  @Test
+  public void testConstructorWithName() {
+    Analysis a = new Analysis("analysis");
+    assertThat(a.getName(), equalTo("analysis"));
+    assertTrue(a.getQuestions().isEmpty());
+  }
+
+  @Test
+  public void testBasicFunctions() {
+    Map<String, String> questions = Collections.singletonMap("question", "questionContent");
+    Analysis a = new Analysis("analysis", questions);
+    assertThat(a.getName(), equalTo("analysis"));
+    assertThat(a.getQuestions(), equalTo(questions));
+    a.setName("other-analysis");
+    assertThat(a.getName(), equalTo("other-analysis"));
+    Map<String, String> otherQuestions =
+        Collections.singletonMap("other-question", "other question content");
+    a.setQuestions(otherQuestions);
+    assertThat(a.getQuestions(), equalTo(otherQuestions));
+  }
+
+  @Test
+  public void testAddQuestion() {
+    Analysis a = new Analysis("analysis");
+    a.addQuestion("question", "questionContent");
+    assertThat(a.getQuestions().get("question"), equalTo("questionContent"));
+    _thrown.expect(BatfishException.class);
+    _thrown.expectMessage(equalTo("Question 'question' already exists for analysis 'analysis'"));
+    a.addQuestion("question", "questionContent");
+  }
+
+  @Test
+  public void testDeleteQuestion() {
+    Map<String, String> questions = new HashMap<>();
+    questions.put("question", "questionContent");
```

Comment:
how come here you use new HashMap with put, but before you used Collections.singletonMap?